### DISCUSSION TITLE
how to make blocks show up with a button

### DISCUSSION THREAD

**Discussion by PolygonPenguin** [TurboWarp/extensions#1441] [Q&A] [answered]
i want to have an advanced button that users can press to get more, more advanced, blocks

**Comment by CST1229** [comment, marked as answer]
Look at the code used in the Animated Text extension. Basically, set the hideFromPalette property of blocks to a value based on a variable, and have a Scratch.BlockType.BUTTON to toggle that variable.

**Comment by penta-quark-neutro** [comment]
example
{func:'herram1',blockType:Scratch.BlockType.BUTTON,hideFromPalette:!ops,text:'mostrar Operadores',}
{func:'herram2',blockType:Scratch.BlockType.BUTTON,hideFromPalette:ops,text:'ocultar Operadores',}
herram1(){ops=false;Scratch.vm.extensionManager.refreshBlocks();}
herram2(){ops=true;Scratch.vm.extensionManager.refreshBlocks();}
